app/knights_info/knights.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+from app.knights_info.armor import Armour
+from app.knights_info.weapon import Weapon
+from app.knights_info.potion import Potion
+
+
+class Knight:
+
+    def __init__(self,
+                 name: str,
+                 power: int,
+                 hp: int,
+                 armour: list,
+                 weapon: dict,
+                 potion: dict = None) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+        self.armour = [Armour(armour_info["part"],
+                              armour_info["protection"])
+                       for armour_info in armour]
+        self.weapon = Weapon(weapon.get("name"), weapon.get("power"))
```

Comment:
U can just unpack the dict `weapon` here. You anyway will have an error during creting a Weapon instance if, for example, `weapon.get("name")` will give a `None`. So you`re not predicting an error with `get` method. Thats why you can just unpack the dict